### DISCUSSION TITLE
Disable nesting with some data properties?

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#1767] [7. Bugs (archived)] [answered]
The new nesting problem is causing some issues, for example, check this code:

```js
<div x-data="{ username: 'what ever' }">

    <div x-data="{ username: null }" @acustomevent="(e) => { username = e.detail.username }">
    
        <input @change="() => { username = value }">

    </div>

</div>
```

Because we redefine username again in the child component, the first username will lose its value.

I understand that this is intended and a new feature, but can we introduce a new feature like private data properties and stop sharing them with child components? so this way, we don't need to always use unique names.

This is especially the case when you use a different file for each component.

for example, we can have `_username` and the alpine should not send it to the child components.

**Comment by SimoTod** [comment, marked as answer]
The internal property doesn't override the external one, see https://codepen.io/SimoTod/pen/eYWdwwW (the test button will always print the value of the external property).

**Reply by ghost** [reply]
The behavior I am facing is because of events.
I will create another issue to describe the problem.
